### PR TITLE
allow multiple images pull

### PR DESCRIPTION
**Pull request by michelpromonet**
Hi,

Please find this PR that intend to pull multiple images and save it to a tar file that could be loaded by docker load.

It is a quick & dirty try, let me know if you think this feature extension is interesting or if there is an other way to do this with the cli.

Best Regards,
Michel.

### CODE REVIEW

**Comment by michelpromonet** [comment]
@googlebot I signed it!

**Comment by jonjohnsonjr** [comment]
Thanks for the PR!

> It is a quick & dirty try, let me know if you think this feature extension is interesting or if there is an other way to do this with the cli.

Definitely interesting and something I'd like to support. CLI UX looks good to me, but I'm a little reluctant to modify the signature for `crane.Save` etc.

Do you mind instead adding three new functions?

* `crane.MultiSave`
* `crane.MultiSaveLegacy`
* `crane.MultiSaveOCI`

Then we can rewrite `Save{Legacy,OCI}` as trivial implementations that just call into `MultiSave...` without breaking anything that depends on the `Save` functions today.

File: pkg/crane/pull.go [hunk 1]
```diff
@@ -91,18 +110,31 @@ func SaveLegacy(img v1.Image, src, path string) error {
...
+	var err error
+	var p layout.Path
+	for _, img := range imgMap {
+		p, err = layout.FromPath(path)
+		if err != nil {
+			p, err = layout.Write(path, empty.Index)
+			if err != nil {
+				return err
+			}
+		}
```
Comment (jonjohnsonjr) [inline pkg/crane/pull.go:133]: We should be able to just do this once outside of the loop.

File: pkg/crane/pull.go [hunk 2]
```diff
@@ -91,18 +110,31 @@ func SaveLegacy(img v1.Image, src, path string) error {
...
+		err = p.AppendImage(img)
```
Comment (jonjohnsonjr) [inline pkg/crane/pull.go:134]: I'd just:

```
if err := p.AppendImage(img); err != nil {
  return err
}
```

File: pkg/crane/pull.go [hunk 3]
```diff
@@ -91,18 +110,31 @@ func SaveLegacy(img v1.Image, src, path string) error {
...
 		if err != nil {
 			return err
 		}
 	}
-	return p.AppendImage(img)
+	return err
```
Comment (jonjohnsonjr) [inline pkg/crane/pull.go:139]: With above suggestions, if we make it out of the loop, just `return nil`.

**Review by jonjohnsonjr — COMMENTED** [review]
LGTM beyond a couple small nits.

Can you also add godoc comments to the Multi funcs?

**Comment by michelpromonet** [comment]
Hi jonjohnsonjr,

Please find update adding comment on MultiWrite method and rework on MultiWriteOCI.

Best Regards,
Michel.

**Review by jonjohnsonjr — APPROVED** [review]
LGTM, thanks!

**Pull request merged**